newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js

```diff
@@ -46,103 +49,135 @@ const useFetchLeaderboards = () => {
-                      }`}
-                    />
-                  ))
-                : null}
-            </SelectField>
-          ) : (
-            <GenericExpressionField
-              expressionType="string"
-              {...props}
-              onChange={onChangeTextValue}
-              onExtractAdditionalErrors={(
-                currentExpression: string,
-                currentExpressionNode: gdExpressionNode
-              ) => {
-                if (!leaderboards)
-                  return `Unable to fetch leaderboards as you are offline.`;
-              }}
-            />
-          )
-        }
-        renderButton={style => (
+    return (
+      <I18n>
+        {({ i18n }) => (
           <>
-            <RaisedButtonWithSplitMenu
-              icon={<OpenInNew />}
-              style={style}
-              primary
-              onClick={() => setIsAdminOpen(true)}
-              buildMenuTemplate={i18n => [
-                {
-                  label: isTextInput
-                    ? i18n._(t`Switch to expression`)
-                    : i18n._(t`Switch to select`),
-                  disabled: !leaderboards,
-                  click: () => setIsTextInput(!isTextInput),
-                },
-              ]}
+            <TextFieldWithButtonLayout
+              renderTextField={() =>
+                !isTextInput ? (
+                  <SelectField
+                    value={props.value}
+                    onChange={onChangeSelectValue}
+                    margin={props.isInline ? 'none' : 'dense'}
+                    fullWidth
+                    floatingLabelText={fieldLabel}
+                    hintText={t`Choose a leaderboard`}
+                    helperMarkdownText={
+                      leaderboards && leaderboards.length === 0
+                        ? i18n._(
+                            t`There are currently no leaderboards created for this game. Open the leaderboards manager to create one.`
+                          )
+                        : null
+                    }
+                  >
+                    {leaderboards && !!leaderboards.length
+                      ? leaderboards.map(leaderboard => (
+                          <SelectOption
+                            key={leaderboard.id}
+                            value={`"${leaderboard.id}"`}
+                            primaryText={`${leaderboard.name} ${
+                              leaderboard.id
+                                ? `(${breakUuid(
+                                    leaderboard.id.substring(0, 8)
+                                  )})`
+                                : ''
+                            }`}
+                          />
+                        ))
+                      : [
+                          <SelectOption
+                            disabled
+                            key="empty"
+                            value="empty"
+                            primaryText={''}
```

Comment:
Should we put a primaryText like "No leaderboards created - create one in the leaderboard manager"?

Reply:
It felt weird to me to have 2 disabled options in the select, I thought it was clear enough when there is no option + the helper text below.
But maybe that's just me.

<img width="852" alt="image" src="https://user-images.githubusercontent.com/32449369/160649332-839b6fb6-69a9-482b-b920-67d35717faec.png">

Reply:
Right - can we replace "Choose a leaderboard" by  "No leaderboards [..]"?

Reply:
Oh yes, good idea!